### DISCUSSION TITLE
Edge Functions: examples for non-Next.js projects

### DISCUSSION THREAD

**Discussion by brycewray** [vercel/community#276] [Idea]
The [recent announcement of Vercel Edge Functions](https://nextjs.org/blog/next-12#introducing-middleware) said they’d be available for any Vercel-based project, regardless of whether it used Next.js. However, as of this writing, the [examples provided](https://github.com/vercel/examples/tree/main/edge-functions) are entirely Next-based, each beginning with code such as:

```js
import type { NextRequest } from 'next/server'

export function middleware(req: NextRequest) {
```

I know they’re based on Cloudflare Workers, but my early attempts to test them with CFW-specific code haven’t worked. Surely would like to see one or two examples, even if only simple ones, that do ***not*** require Next.js, so I could try to adapt CFW code I’m currently using.

Thanks in advance to anyone who could provide such non-Next examples of code for Vercel Edge Functions.

**Comment by brycewray** [comment]
(Not an answer, obviously; just a “bump, bump,” so to speak.)

Any thoughts on an ETA for even a few, barest-bones versions of these? @leerob or @mcsdevv ?

(Now marked as “answer” due to @mcsdevv reply below.)

**Reply by mcsdevv** [reply]
Hey @brycewray, we're releasing some changes in the coming days which will allow for the use of middleware with any framework!

**Reply by brycewray** [reply]
Excellent! Thanks for the advisory.

**Reply by leerob** [reply]
Yep, we're working on adding more example to https://github.com/vercel/examples

**Reply by brycewray** [reply]
Wonderful. Thanks!

**Reply by ptrhck** [reply]
When can we expected this to be released? Very much looking forward to this!

**Reply by willwillems** [reply]
As of today all the examples are still next specific

**Comment by khrome83** [comment]
The File System API exposes middleware support for Edge Functions. 

The example in the docs are not based on Next.js and available to anyone.

https://vercel.com/docs/file-system-api#resources/middleware

You need to set the "ENABLE_FILE_SYSTEM_API" and generate the .output folder.

I don't know how well this will play, since chances are you are also moving other items over to the .output file or your pulling them in with the file resources. 

I am still playing with the ENABLE_FILE_SYSTEM_API to try and build out ISR. And have not yet played with the middleware itself.

**Reply by eric-burel** [reply]
The new link seems to be https://vercel.com/docs/build-output-api/v3#features/edge-middleware, not sure yet how to set that up in a non-Next.js app though it's a bit complex

**Reply by leerob** [reply]
Planning to post more guidance on this soon with examples!
